src/persistence/subject-builder/OneToManySubjectBuilder.ts

```diff
@@ -165,13 +165,22 @@ export class OneToManySubjectBuilder {
                 const removedRelatedEntitySubject = new Subject({
                     metadata: relation.inverseEntityMetadata,
                     parentSubject: subject,
-                    canBeUpdated: true,
                     identifier: removedRelatedEntityRelationId,
-                    changeMaps: [{
+                });
+
+                // in case the inverse relation is primary we cannot remove the relation by setting the
+                // foreign key field to null. Instead remove the entity.
+                if (relation.inverseRelation
```

Comment:
WE CANNOT SIMPLY REMOVE INVERSE ENTITY IF CASCADE WERE NOT DEFINED. Where cascade remove check is?

Reply:
also, this case only covers when foreign key of entity is a primary key, and does not cover simple cases. Cascade remove logic should not be tight to the fact of foreign key column being primary or not.